docs/integrations/nevermined.md

```diff
@@ -0,0 +1,337 @@
+# Integrating Nevermined with Olas
+
+This guide explains how to integrate Nevermined's subscription and payment services with Olas autonomous agents and services. Nevermined provides monetization capabilities, subscription management, and usage accounting for AI agents in the Olas ecosystem.
+
+## What is Nevermined?
+
+Nevermined is a data ecosystem that enables subscription-based services within the Olas network. It provides:
+
+- Monetization capabilities for AI agents
+- Usage monitoring and accounting
+- Subscription management (time-based or request-based)
+- Payment processing (both fiat and crypto)
+
+## Integration Options
+
+There are multiple ways to integrate Nevermined with Olas:
+
+### 1. Basic Integration (No Code Modification)
+
+For simple monetization without advanced features:
+
+1. **Register your Olas agent in the Nevermined App**
+2. **Define pricing and access models**
+
+This approach is suitable if you only need:
+
+- Basic monetization
+- Simple usage accounting
+- Time or credit-based access control
+
+### 2. Advanced Integration (Using Payment Libraries)
+
+For more sophisticated features, integrate the Nevermined Payment Libraries into your Olas agent:
+
+- Purchase access to other agents
+- Query other agents using the Nevermined Query Protocol
+- Implement dynamic credit charging based on request complexity
+
+## Step-by-Step Integration Guide
+
+### Prerequisites
+
+- An existing Olas agent or service
+- Python >= 3.10 (for Python implementations)
+- Node.js >= 16 (for TypeScript/JavaScript implementations)
+- A Nevermined account
+- (Optional) A Stripe account for fiat payments
+
+### Basic Integration (No Code Changes)
+
+1. **Register your Olas agent with Nevermined**
+
+   Visit the [Nevermined App](https://app.nevermined.io/) and:
+   
+   - Create an account
+   - Select "Register AI Agent"
+   - Provide your agent's HTTP endpoint
+   - Generate and securely store a JWT authentication token
+   - Assign a unique identifier to your service
+
+2. **Create a Payment Plan**
```

Comment:
In the app the creation of the Agent and the Plan is done in the same step. Can be done in 2 steps but in that case the Plan needs to be created first.